src/Orleans.Runtime/Silo/SiloHost.cs

```diff
@@ -114,7 +114,9 @@ public void InitializeOrleansSilo()
         /// </summary>
         public void UnInitializeOrleansSilo()
         {
-            Utils.SafeExecute(UnobservedExceptionsHandlerClass.ResetUnobservedExceptionHandler);
+            var unObservedExceptionHandler = this.orleans?.Services?.GetService<UnobservedExceptionsHandler>();
+            if(unObservedExceptionHandler != null)
+                Utils.SafeExecute(unObservedExceptionHandler.Dispose);
```

Comment:
This is not needed, since the Silo itself would dispose it if it was initialized by it.
I guess we can remove this `UnInitializeOrleansSilo` method entirely, but maybe we can preserve in case we need it in the future. TBH, we are probably killing SiloHost soon anyway, so that might not be a compelling argument. I don't really care either way.

Reply:
lol. You comments rhymed. I will keep the UnInitializeOrleansSilo method for backward-compatible reason and maybe we want it in the future.